hist/hist/src/TMultiDimFit.cxx

```diff
@@ -2120,8 +2120,8 @@ void TMultiDimFit::MakeRealCode(const char *filename,
    << "        for (k = 3; k <= power; k++) { " << std::endl
    << "          p3 = p2 * v;" << std::endl;
    if (fPolyType == kLegendre)
-      outFile << "          p3 = ((2 * i - 3) * p2 * v - (i - 2) * p1)"
-      << " / (i - 1);" << std::endl;
+      outFile << "          p3 = ((2 * k - 3) * p2 * v - (k - 2) * p1)"
+      << " / (k - 1);" << std::endl;
    if (fPolyType == kChebyshev)
```

Comment:
Are we sure in case of Chebyshev the code is correct? 
There is there no dependency on `k`

Reply:
In the recurrent Chebyshev formulation, there is no need to specify `k`, the loop depth encodes it:

![image](https://github.com/root-project/root/assets/10653970/38d3cdd3-f239-42dc-9eee-b174cd567c44)

In contrast, for Legendre, you need `k` explicitly:

![image](https://github.com/root-project/root/assets/10653970/e0ec446b-5cbe-4d84-a531-52d93191638f)

Reply:
Thank you @ferdymercury  for checking this!
I agree, the code looks correct now!